REACT_STYLEGUIDE.MD

```diff
@@ -0,0 +1,194 @@
+Good:
+```
+export class ClickCounter extends React.Component {
+  constructor() {
+    this.state = {
+      clickCount: 0
+    };
+  }
+
+  onClick() {
+    this.setState({ clickCount: this.state.clickCount++ });
+  }
+
+  render({ onClick }) {
+    return <button className="kuiButton" onClick={this.onClick}/>
+  }
+}
+```
+Bad:
+```
+ export const ClickCounter = React.createClass({
+  getInitialState() {
+    return {
+      clickCount: 0
+    };
+  },
+  onClick() {
+    this.setState({ clickCount: this.state.clickCount++ });
+  },
+  render({ onClick }) {
+    return <button className="kuiButton" onClick={this.onClick}/>
+  }
+});
+```
+
+### Prefer reactDirective over react-component
+reactDirective and react-component are two different ways of embedding react in angular. Using `react-component` means adding a bunch of components into angular, while `reactDirective` keeps them isolated, and is also a more sucinct syntax.
+
+Good:
+```
+<hello-component fname="person.fname" lname="person.lname" watch-depth="reference"></hello-component>
+```
+Bad:
+```
+<react-component name="HelloComponent" props="person" watch-depth="reference"/>
+```
+
+### Prefix ui_framework elements with kui, but not their file names.
+Good:
+```
+button.js:
+export function KuiButton({ onClick, isDisabled }) {
+  return <button className="kuiButton" onClick={onClick} isDisabled={isDisabled}/>
+};
+```
+Bad:
+```
+button.js:
+export class Button extends React.Component {
+  render({ onClick, isDisabled }) {
+    return <button className="kuiButton" onClick={onClick} isDisabled={isDisabled}/>
+  }
+}
+```
+The filenames leave it off because snake casing already increases file name length. The prefix is left on for the components to avoid any naming collisions.
+
+### Action function names and prop function names
+
+Name action functions in the form of a strong verb and passed properties in the form of on<Subject><Change>. E.g:
+```
+<sort-button onClick={action.sort}/>
+<pagerButton onPageNext={action.turnToNextPage} />
+```
+
+### Avoid creating a function and passing that as a property, in render functions.
+Best (relies on [stage 2 proposal](https://github.com/tc39/proposal-class-public-fields)):
+```
+export class ClickCounter extends React.Component {
+  constructor() {
+    this.state = { clickCount: 0 };
+  }
+
+  // This syntax ensures `this` is bound within handleClick
+  onClick = () => {
+    this.setState({ clickCount: this.state.clickCount++ });
+  }
+
+  render({ onClick }) {
+    return <button className="kuiButton" onClick={this.onClick}/>
+  }
+}
+```
+Good:
+```
+export class ClickCounter extends React.Component {
+  constructor() {
+    this.state = { clickCount: 0 };
+    this.onClick = this.onClick.bind(this);
+  }
+
+  onClick() {
+    this.setState({ clickCount: this.state.clickCount++ });
+  }
+
+  render({ onClick }) {
+    return <button className="kuiButton" onClick={this.onClick}/>
+  }
+}
+```
+Okay:
```

Comment:
In which way is this more "okay" than the bad example below? `Function.prototype.bind` also creates a new function on every render.